docs/assemblies/assembly-operator-api.adoc

```diff
@@ -0,0 +1,44 @@
+// file included in the following:
+//
+// kroxylicious-operator/index.adoc
+
+[id='assembly-operator-api-{context}']
+= API concepts
+
+== API resources used by the Kroxylicious Proxy
+
+The operator takes these custom resources and core Kubernetes resources as inputs:
+
+`KafkaProxy`::
+Defines an instance of the proxy.
+`VirtualKafkaCluster`::
+Represents a logical Kafka cluster that will be exposed to Kafka clients.
+`KafkaProxyIngress`::
+Configures how a virtual cluster is exposed on the network to Kafka clients.
+`KafkaService`::
+Specifies a _backend_ Kafka cluster for a virtual cluster.
+`KafkaProtocolFilter`::
+Specifies filter mechanisms for use with a virtual cluster.
+`Secret`::
+`KafkaService` and `KafkaProtocolFilter` resources may reference a `Secret` to provide security-sensitive data such as TLS certificates or passwords.
```

Comment:
The work of #2081 will mean KafkaVirtualCluster should be added to this list.

Reply:
I think #2081 will be a whole other PR.